### PR TITLE
v9 -> v10

### PR DESCRIPTION
**Pull request by sandervspl**

### CODE REVIEW

File: next.config.ts [hunk 1]
```diff
@@ -46,105 +44,105 @@ const config = (phase: string, config) => {
...
-        if (config.resolve) {
-          if (Array.isArray(config.resolve.plugins)) {
-            config.resolve.plugins.push(new TSConfigPathsPlugin());
-          } else {
-            config.resolve.plugins = [new TSConfigPathsPlugin()];
-          }
+        if (!config.resolve) {
+          config.resolve = {
+            plugins: [],
+          };
```
Comment (rnnyrk) [inline next.config.ts:138]: Why is this necessary when it's empty and te same as line 119?

File: patches/next-redux-wrapper+6.0.2.patch [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+     getServerSideProps: <P extends {} = any>(callback: (context: GetServerSidePropsContext & {
+-        store: Store<S, A>;
++        store: S;
+     }) => void | P) => GetServerSideProps<P, ParsedUrlQuery>;
+     getStaticProps: <P_1 extends {} = any>(callback: (context: GetStaticPropsContext & {
+-        store: Store<S, A>;
++        store: S;
+     }) => void | P_1) => GetStaticProps<P_1, ParsedUrlQuery>;
+     withRedux: (Component: any) => React.FunctionComponent<WrapperProps> & {
+         getInitialProps?(context: NextPageContext<any, AnyAction>): WrapperProps | Promise<WrapperProps>;
```
Comment (rnnyrk) [inline patches/next-redux-wrapper+6.0.2.patch:27]: So this will be a permanent patch?

File: scripts/watchPagesFolder.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      }
+    });
+
+    watcher.on('error', (err) => {
+      reject(err);
+    });
+  });
+}
+
+module.exports = watchPagesFolder;
```
Comment (rnnyrk) [inline scripts/watchPagesFolder.js]: Can you explain this new scripts folder?

File: tsconfig.nextconfig.json [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
     "esModuleInterop": true,
     "allowSyntheticDefaultImports": true,
     "resolveJsonModule": true,
-    "module": "commonjs",
-    "outDir": ".",
-    "target": "es5",
-    "lib": [
-      "es2017"
-    ]
+    "removeComments": true,
```
Comment (rnnyrk) [inline tsconfig.nextconfig.json:10]: Why is `removeComments` here `true`, compared to react-prime where you explicitly set it to `false`?

File: next.config.ts (hunk 1 above)
Comment (sandervspl) [reply next.config.ts:138]: You can have a `config` object without the `resolve` property. In the following lines I spread `config.resolve.plugins` into a new array, so I need to make sure `config.resolve.plugins` exists and is an array otherwise the code could error. So I need to initialize `config.resolve.plugins` with an empty array if it does not exist yet.

File: patches/next-redux-wrapper+6.0.2.patch (hunk 2 above)
Comment (sandervspl) [reply patches/next-redux-wrapper+6.0.2.patch:27]: I can make a pull-request for the package, but for now this might be permanent, yes.

File: scripts/watchPagesFolder.js (hunk 3 above)
Comment (sandervspl) [reply scripts/watchPagesFolder.js]: In the past we used `next-routes` to create a routes object. We could then infer a `Routes` type from this object to make routing type-checking possible. I removed the `next-routes` package because it became obsolete due to Next adding dynamic routes support. This means we no longer have to update and keep track of a routes object, which also means we lose type-checking on routes.

This specific scripts creates a watcher for the `pages` folder and waits for updates on the folder structure. From this it will automatically create route types that you can see here https://github.com/react-prime/react-prime-ssr/blob/v10/src/types/routes.ts

File: tsconfig.nextconfig.json (hunk 4 above)
Comment (sandervspl) [reply tsconfig.nextconfig.json:10]: This is the tsconfig for the nextconfig file, not the source code

File: scripts/watchPagesFolder.js (hunk 3 above)
Comment (sandervspl) [reply scripts/watchPagesFolder.js]: I have removed these auto-generating scripts for now. They require more testing in an actual project to see what edge cases come up.

Comment (rnnyrk) [reply scripts/watchPagesFolder.js]: Ah okay, I was wondering why you'd need this besides HMR. The auto generating types explains :)

**Comment by rnnyrk** [comment]
Shall I merge this one, Sander?

**Comment by sandervspl** [comment]
Yes this can be merged
